cloudevents/emitter_test.go

```diff
@@ -0,0 +1,132 @@
+package cloudevents
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/google/go-cmp/cmp"
+)
+
+func makeClient(t *testing.T) (c client.Client, p *http.Protocol) {
+	p, err := http.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err = client.New(p)
+	if err != nil {
+		t.Errorf("failed to make client %s", err.Error())
+	}
+	return
+}
+
+func receiveEvents(t *testing.T, ctx context.Context, events chan<- event.Event) {
+	c, _ := makeClient(t)
+	go func() {
+		err := c.StartReceiver(ctx, func(ctx context.Context, event event.Event) error {
+			go func() {
+				events <- event
+			}()
+			return nil
+		})
+		if err != nil {
+			t.Errorf("failed to start receiver %s", err.Error())
+		}
+	}()
+	time.Sleep(1 * time.Second) // let the server start
+}
+
+func TestEmitterDefaults(t *testing.T) {
+	events := make(chan event.Event)
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	// start a cloudevent client that receives events
+	// and sends them to a channel
+	receiveEvents(t, ctx, events)
+
+	emitter := NewEmitter()
+	emitter.Emit("http://localhost:8080")
```

Comment:
I'm not sure if this is exposed by the `cloudevent` library, but creating a net listener without a port specified allows the OS to choose the port, the final address of the listening service is then available using the `.Addr()` accessor.  This gets around the all-too-common problem of there already being a service attached to `:8080` on the box running the tests.